### ISSUE TITLE
Querying document fails when the criteria includes an embedded array field with sub-documents

### ISSUE THREAD

**Issue by catchshyam** [Automattic/mongoose#5501]
I think this is a regression in the latest version of Mongoose (4.11.3). The following scenario is working fine with mongoose v4.4.12.

### Schema:
```
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var MemberSchema = new Schema({
    _id: false,
    teamId: {type: Schema.Types.ObjectId},
    batchId: {type: Schema.Types.ObjectId}
});

var GroupSchema = new Schema({
    members: [MemberSchema],
    status: String
});

mongoose.model('Group', GroupSchema);

```

### Sample Document:

> {
>     "_id" : ObjectId("597620fcba76f211586c9203"),
>     "status" : "new",
>     "members" : [{
>             "teamId" : ObjectId("597620fcba76f211586c9201"),
>             "batchId" : ObjectId("597620fcba76f211586c9202")
>       }],
>     "__v" : 0
> }
> 

### Sample Application Code:

```
mongoose.set('debug', true);
var Group = mongoose.model('Group');
Group
    .findById("597620fcba76f211586c9203", "_id members")
    .lean()
    .exec()
    .then(function (group) {
        Group.update(group, {$set: {status: "approved"}}).exec().then(function (result) {
            console.log("Group update result ", result);
        });
    });

```
### Debug Output in Mongoose v4.11.3:

Mongoose: groups.update({ members: [ { teamId: ObjectId("597620fcba76f211586c9201"), batchId: ObjectId("597620fcba76f211586c9202") } ], _id: ObjectId("597620fcba76f211586c9203") }, { '$set': { status: 'approved' } }, {})
Group update result  { ok: 1, **nModified: 0**, n: 0 }

### Output in Mongoose v4.4.12:

Mongoose: groups.update({ members: [ { teamId: ObjectId("597620fcba76f211586c9201"), batchId: ObjectId("597620fcba76f211586c9202") } ], _id: ObjectId("597620fcba76f211586c9203") }) { '$set': { status: 'approved' } } {} 
Group update result  { ok: 1, **nModified: 1**, n: 1 }

Currently I can not upgrade my Mongoose version from 4.4.12 since I use the criteria with array sub-documents in a lot of places in the application. It would also help if there are any short term workaround.

My Node version is 4.2.2.

Thanks,
Shyam

**Comment by antolovich** [comment]
This seems to have been fixed in Mongoose v4.11.4.

Output in Mongoose v4.11.3:

```Mongoose: groups.findOne({ _id: ObjectId("597620fcba76f211586c9203") }, { fields: { _id: 1, members: 1 } })
Mongoose: groups.findOne({ _id: ObjectId("597620fcba76f211586c9203") }, { fields: { _id: 1, members: 1 } })
Mongoose: groups.update({ members: [ { teamId: ObjectId("597620fcba76f211586c9201"), batchId: ObjectId("597620fcba76f211586c9202") } ], _id: ObjectId("597620fcba76f211586c9203") }, { '$set': { status: 'approved' } }, {})
Group update result  { n: 0, nModified: 0, ok: 1 }

```

Output in Mongoose v4.11.4:
```
Mongoose: groups.findOne({ _id: ObjectId("597620fcba76f211586c9203") }, { fields: { _id: 1, members: 1 } })
Mongoose: groups.update({ members: [ { batchId: ObjectId("597620fcba76f211586c9202"), teamId: ObjectId("597620fcba76f211586c9201") } ], _id: ObjectId("597620fcba76f211586c9203") }, { '$set': { status: 'approved' } }, {})
Group update result  { n: 1, nModified: 1, ok: 1 }```

**Comment by catchshyam** [comment]
Yes fixed in v4.11.4. Thank you.

**Comment by catchshyam** [comment]
@antolovich, there are other scenarios in which I could reproduce this issue. But it looks like the culprit is MongoDB server. FYI, here is the mongodb bug report https://jira.mongodb.org/browse/SERVER-30392
